IPython/kernel/zmq/kernelbase.py

```diff
@@ -347,41 +347,43 @@ def execute_request(self, stream, ident, parent):
             self.log.error("Got bad msg: ")
             self.log.error("%s", parent)
             return
-        
-        stop_on_error = content.get('stop_on_error', True)
 
         md = self._make_metadata(parent['metadata'])
-        
+
         # Re-broadcast our input for the benefit of listening clients, and
         # start computing output
         if not silent:
             self.execution_count += 1
             self._publish_execute_input(code, parent, self.execution_count)
-        
+
         reply_content = self.do_execute(code, silent, store_history,
                                         user_expressions, allow_stdin)
+        self.send_execute_reply(stream, ident, parent, md, reply_content)
+
+    def send_execute_reply(self, stream, ident, parent, md, reply_content):
+        """ Send a reply to execute_request """
 
         # Flush output before sending the reply.
-        sys.stdout.flush()
-        sys.stderr.flush()
-        # FIXME: on rare occasions, the flush doesn't seem to make it to the
-        # clients... This seems to mitigate the problem, but we definitely need
-        # to better understand what's going on.
-        if self._execute_sleep:
-            time.sleep(self._execute_sleep)
+        self._flush_execute_output()
 
-        # Send the reply.
+        # Cleanup the reply and prepare metadata
         reply_content = json_clean(reply_content)
-        
+
+        md = self._make_metadata(md)
```

Comment:
By `md = self._make_metadata(md)` I meant `md = md.copy()` (the effect is the same) - a way to avoid mutating dictionary passed as an argument. it is not the best coding :) 

If 'started' field is *not* calculated during the first `_make_metadata()` call then `md` argument is not needed, `md` can be a local varaible.